app/models/service_template.rb

```diff
@@ -103,12 +88,31 @@ def self.with_additional_tenants
     references(table_name, :tenants).includes(:service_template_tenants => :tenant)
   end
 
+  def self.all_catalog_item_types
+    @all_catalog_item_types ||= begin
+      builtin_catalog_item_types = {
+        "generic"               => N_("Generic"),
+        "generic_orchestration" => N_("Orchestration"),
+      }
+
+      ExtManagementSystem.supported_types_for_catalog
+        .flat_map(&:catalog_types)
+        .reduce(builtin_catalog_item_types, :merge)
+    end
+  end
+
   def self.catalog_item_types
-    ci_types = Set.new(Rbac.filtered(ExtManagementSystem.all).flat_map(&:supported_catalog_types))
+    ci_types = Set.new(
+      Rbac.filtered(ExtManagementSystem.all)
```

Comment:
would love if `Rbac.filtered` instead became a scope called `.rbac_filtered`, so you would do `ExtManagementSystem.rbac_filtered.yaddayadda`

Reply:
@Fryguy please don't move rbac into the individual classes.
We need rbac to work at the parent class layer.

When we subclass and change the privileges or change the relations, then our filtering (which only sees the top level class) will get it wrong.

For OOP reasons, it totally makes sense, but since we are running rbac as a queries only, we need to be able to make decisions at the top class level only.

Introducing rbac into the objects means we'll have to fetch every object to see if it passes. Which is pretty bad once an installation gets big enough